.github/workflows/ci.yml

```
@@ -216,13 +216,22 @@ jobs:
           cat ./.github/workflows/resources/poseidon-minimal.service | envsubst > ~/.config/systemd/user/poseidon.service
           systemctl --user daemon-reload
           systemctl --user start poseidon.socket
+        if: ${{ success() || failure() }}
       - name: Print Poseidon Failure logs
         if: failure()
-        run: journalctl -xen --no-pager
+        run: journalctl --user -xen --no-pager
       - name: Run e2e recovery tests
+        run: make e2e-test-recovery
+        if: ${{ success() || failure() }}
+      - name: Print Systemd Failure logs
         run: |
-          tail -f /var/log/syslog &
-          make e2e-test-recovery
+          /usr/bin/systemctl --user show poseidon.service -p NRestarts
+          journalctl --user -xe -u poseidon.service --no-pager
+        if: failure()
+      - name: Print Coverage files
+        run: |
+          killall poseidon
```

Wouldn't it make sense to stop Poseidon through systemd?

```suggestion
          systemctl --user stop poseidon.service poseidon.socket
```

I think, using `killall` is fine as well (that uses SIGTERM by default), but the other one would probably be similar to the actual usage (we are, nevertheless, still in an e2e context).